cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt

```diff
@@ -69,67 +65,76 @@ internal class ServiceImpl(
-        for (i in 0 until this.length()) {
-            list.add(this.getString(i))
-        }
-        return list
-    }
-
-    override fun sendCustomConsentServ(customConsentReq: CustomConsentReq, env: Env): Either<GdprCS> = check {
+    override fun sendCustomConsentServ(
+        consentUUID: String,
+        propertyId: Int,
+        vendors: List<String>,
+        categories: List<String>,
+        legIntCategories: List<String>
+    ): Either<GdprCS> = check {
         if (connectionManager.isConnected.not()) throw NoInternetConnectionException()
 
-        networkClient.sendCustomConsent(customConsentReq, env)
-            .map {
-                if (campaignManager.gdprConsentStatus == null) {
-                    genericFail("CustomConsent cannot be executed. Consent is missing!!!")
-                }
-
-                val categories: List<String> = (it.content.get("categories") as JSONArray).toArrayList()
-                val vendors: List<String> = (it.content.get("vendors") as JSONArray).toArrayList()
-                val legIntCategories: List<String> = (it.content.get("legIntCategories") as JSONArray).toArrayList()
-                val specialFeatures: List<String> = (it.content.get("specialFeatures") as JSONArray).toArrayList()
-
-                val grantsString: String = (it.content.get("grants") as JSONObject).toString()
-                val grants = JsonConverter.converter.decodeFromString<Map<String, GDPRPurposeGrants>>(grantsString)
-                val updatedGrants = campaignManager.gdprConsentStatus?.copy(
-                    grants = grants,
-                    categories = categories,
-                    vendors = vendors,
-                    legIntCategories = legIntCategories,
-                    specialFeatures = specialFeatures
-                )
-                campaignManager.gdprConsentStatus = updatedGrants
+        try {
+            val response = networkClient.sendCustomConsent(
+                consentUUID = consentUUID,
+                propertyId = propertyId,
+                vendors = vendors,
+                categories = categories,
+                legIntCategories = legIntCategories
+            )
+            if (campaignManager.gdprConsentStatus == null) {
+                genericFail("CustomConsent cannot be executed. Consent is missing!!!")
             }
-        campaignManager.gdprConsentStatus!!
+            val grants = response.grants.map {
+                    e -> e.key to GDPRPurposeGrants(e.value.vendorGrant,e.value.purposeGrants)
+            }.toMap()
+            campaignManager.gdprConsentStatus = campaignManager.gdprConsentStatus?.copy(
+                grants = grants,
+                categories = response.categories,
+                vendors = response.vendors,
+                legIntCategories = response.legIntCategories,
+                specialFeatures = response.specialFeatures
+            )
+            campaignManager.gdprConsentStatus!!
```

Comment:
What's the purpose of this unwrapping in line 98?
```kotlin
campaignManager.gdprConsentStatus!!
```

If you want to return it, it's better to do it explicitly with the `return` keyword.